### DISCUSSION TITLE
Recommended ROS2 version to use with Universal Robots

### DISCUSSION THREAD

**Discussion by doshininad** [UniversalRobots/Universal_Robots_ROS2_Driver#469] [Q&A] [answered]
The documentation shows that the execution tests are failing with ROS2 Galactic, Humble and Rolling. So, is it recommended to use ROS2 Foxy with Universal Robots?
![Capture](https://user-images.githubusercontent.com/37081073/187763923-1f032f3c-67b6-43a7-857e-f6e42bbc2a38.JPG)

**Comment by destogl** [comment, marked as answer]
Execution test may be failing from various reasons. Nevertheless, the library is building with all version so it should be usable.

I would always work with the newest stable version (humble at the moment) or rolling.

IMO foxy is too obsolete, especially concerning performance and features in core ROS2 libraries and ros2_control.
